src/version.js

```diff
@@ -11,58 +11,44 @@ module.exports = (store) => {
     /**
      * Check if a version file exists.
      *
-     * @param {function(Error, bool)} callback
-     * @returns {void}
+     * @returns {Promise<bool>}
      */
-    exists (callback) {
-      store.has(versionKey, callback)
+    async exists () {
+      return store.has(versionKey)
     },
     /**
      * Get the current version.
      *
-     * @param {function(Error, number)} callback
-     * @returns {void}
+     * @returns {Promise<Integer>}
      */
-    get (callback) {
-      store.get(versionKey, (err, buf) => {
-        if (err) {
-          return callback(err)
-        }
-        callback(null, parseInt(buf.toString().trim(), 10))
-      })
+    async get () {
+      const buf = await this.get(versionKey)
```

Comment:
This line causes a stack overflow, I think it should be
```suggestion
      const buf = await store.get(versionKey)
```

Reply:
It does, I have some pending changes stored locally for this, that I haven't pushed back up yet. Do you mind if I drop a PTAL when I push the various fixes I have going locally, this PR  was a bit tricker to write and is probably more error prone in places, because I wrote it  before the other PRs. Thanks for looking into it!

Reply:
No worries, I just noticed this when testing some code that has this as a dependency. The rest of it looks great to me.